src/compiler/parser.ts

```diff
@@ -3032,11 +3025,17 @@ namespace ts {
                     // do not try to parse initializer
                     return undefined;
                 }
+                if (inParameter && requireEqualsToken) {
+                    // = is required when speculatively parsing arrow function parameters,
+                    // so return a fake initializer as a signal that the equals token was missing
+                    const result = createNode(SyntaxKind.Identifier, scanner.getStartPos()) as Identifier;

```

Comment:
Should probably use `createMissingNode`, just in case, right?

Reply:
Just in case the node slips through into the real parse, right? That incurs some overhead, and I was actually thinking of going the other way; just creating a sentinel Identifier node with no content and just use `===` to identify it later.

Hmmm...I don't know, I'll try the missing node I guess. Better safe than sorry, and the overhead of this parsing pales next to the rest of speculative arrow parsing, which happens every time we see a parenthesis.